WooCommerce/Classes/POS/Card Present Payments/CardPresentPaymentFacade.swift

```diff
@@ -0,0 +1,41 @@
+import Foundation
+import struct Yosemite.Order
```

Comment:
I would prefer that we didn't use this at all, it's leaking an implementation detail that we have to collect payment for an order. Such is reality!

Reply:
Looking at `CollectOrderPaymentUseCase`, it looks like we just need these properties from `Yosemite.Order`:

- total
- currency
- datePaid
- orderID
- siteID (this we don't need to get from the order)
- billingAddress

And in `OrderAction.retrieveOrder`, what's needed from the input `Yosemite.Order` is just the siteID and orderID. We could potentially just pass in an order ID and keep a `Yosemite.Order` after `checkOrderIsStillEligibleForPayment` unless the order properties above are needed before `OrderAction.retrieveOrder` is called.

Reply:
Good thinking. Let's come back to this when we do the collect payment work, and leave it as `Order` for now to get this merged. I've added a note to #12867